staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -75,23 +96,63 @@ func getRoutes(input io.Reader) ([]Route, error) {
 			Interface:   fields[0],
 			Destination: dest,
 			Gateway:     gw,
+			Family:      familyIPv4,
+		})
+	}
+	return routes, nil
+}
+
+func getIPv6DefaultRoutes(input io.Reader) ([]Route, error) {
+	routes := []Route{}
+	scanner := bufio.NewReader(input)
+	for {
+		line, err := scanner.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		fields := strings.Fields(line)
+		dest, err := parseIP(fields[0], familyIPv6)
+		if err != nil {
+			return nil, err
+		}
+		gw, err := parseIP(fields[4], familyIPv6)
+		if err != nil {
+			return nil, err
+		}
+		if !dest.Equal(net.IPv6zero) {
+			continue
+		}
+		if gw.Equal(net.IPv6zero) {
+			continue // loopback
+		}
+		routes = append(routes, Route{
+			Interface:   fields[9],
+			Destination: dest,
+			Gateway:     gw,
+			Family:      familyIPv6,
 		})
 	}
 	return routes, nil
 }
 
-func parseIP(str string) (net.IP, error) {
+func parseIP(str string, family AddressFamily) (net.IP, error) {
 	if str == "" {
 		return nil, fmt.Errorf("input is nil")
 	}
 	bytes, err := hex.DecodeString(str)
 	if err != nil {
 		return nil, err
 	}
-	if len(bytes) != net.IPv4len {
-		return nil, fmt.Errorf("invalid IPv4 address in route")
+	if family == familyIPv4 {
```

Comment:
I disagree about checking the family. We are calling parseIP() with either a **hardcoded** familyIPv4 or familyIPv6 to allow this code to be reused for IPv4 and IPv6, to reduce duplication. It is just a selector - could have been a boolean forIPv4, but I thought it was clearing calling:
```    ip, err := parseIP(x, familyIPv6)```

is clearer than:
```    ip, err := parseIP(x, false)```